client/components/NewProblem/NewProblem.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import './NewProblem.scss';
+
+export class NewProblem extends Component {
+    state = {
+        subject: '',
+        description: ''
+    };
+
+    setSubject = (event) => {
+        this.setState({
+            subject: event.target.value
+        });
+    };
+
+    setDescription = (event) => {
+        this.setState({
+            description: event.target.value
+        });
+    };
+
+    sendProblem = () => {
+        fetch('/api/problems', {
```

Comment:
this stuff should be moved to action creator function
http://redux.js.org/docs/basics/Actions.html#action-creators

ALL BE calls should be in Action creators.
Your react code should just dispatch redux actions.